### PR TITLE
Add mobile number to registration form 

### PR DESCRIPTION
**Pull request by smarzban**

### CODE REVIEW

**Comment by smarzban** [comment]
@miltontony @justinvdm ready for review

File: molo/profiles/admin.py [hunk 1]
```diff
@@ -1,3 +1,28 @@
...
+
+@admin.register(User)
+class ProfileUserAdmin(UserAdmin):
+    list_display = UserAdmin.list_display + (
+        'date_joined', '_alias', '_mobile_number')
+
+    list_filter = UserAdmin.list_filter + ('date_joined', )
+
+    def _alias(self, obj, *args, **kwargs):
+        if hasattr(obj, 'profile') and obj.profile.alias:
```
Comment (justinvdm) [inline molo/profiles/admin.py:21]: When do we not have a `profile` attribute?

File: molo/profiles/forms.py [hunk 2]
```diff
@@ -66,15 +83,17 @@ class EditProfileForm(forms.ModelForm):
...
-        fields = ['alias', 'date_of_birth']
+        fields = ['alias', 'date_of_birth', 'mobile_number']
 
     def clean(self):
         alias = self.cleaned_data.get('alias', None)
         date_of_birth = self.cleaned_data.get('date_of_birth', None)
-        if (alias or date_of_birth):
+        mobile_number = self.cleaned_data.get('mobile_number', None)
+        if (alias or date_of_birth or mobile_number):
             return self.cleaned_data
```
Comment (justinvdm) [inline molo/profiles/forms.py:98]: Doesn't this mean the user only needs to enter one of _either_ an alias, date of birth, or mobile number? I would have expected us to require _all_ fields.

File: molo/profiles/forms.py [hunk 3]
```diff
@@ -66,15 +83,17 @@ class EditProfileForm(forms.ModelForm):
...
     class Meta:
         model = UserProfile
-        fields = ['alias', 'date_of_birth']
+        fields = ['alias', 'date_of_birth', 'mobile_number']
 
     def clean(self):
         alias = self.cleaned_data.get('alias', None)
         date_of_birth = self.cleaned_data.get('date_of_birth', None)
-        if (alias or date_of_birth):
+        mobile_number = self.cleaned_data.get('mobile_number', None)
```
Comment (justinvdm) [inline molo/profiles/forms.py:96]: Noting one of the things we discussed so I don't forget it later: we need to allow the user to not have to enter a number if `mobile_number_required` is true but `show_mobile_number_field` is false.

File: molo/profiles/models.py [hunk 4]
```diff
@@ -2,6 +2,23 @@
...
+    show_mobile_number_field = models.BooleanField(
+        default=False,
+        editable=True,
+        verbose_name=_("Add mobile number field to registration"),
+    )
+    mobile_number_required = models.BooleanField(
+        default=False,
+        editable=True,
+        verbose_name=_('Mobile number required'),
+    )
```
Comment (justinvdm) [inline molo/profiles/models.py:21]: Can we add a TODO comment here (or somewhere else) for greying out `mobile_number_required` in the UI if `show_mobile_number_field` isn't checked.

File: molo/profiles/templates/profiles/viewprofile.html [hunk 5]
```diff
@@ -8,6 +9,9 @@
   <h3>{% trans "Hello " %}{{ user.username }}</h3>
   <label class="label-wide">{% trans "Display Name: " %}{{ user.profile.alias | default:'Anonymous' }}</label>
   <label class="label-wide">{% trans "Date of Birth: " %}{{ user.profile.date_of_birth | date:"M d, Y" }}</label>
+  {% if settings.profiles.UserProfilesSettings.show_mobile_number_field %}
+    <label class="label-wide">{% trans "Mobile Number " %}{{ user.profile.mobile_number }}</label>
+  {% endif %}
```
Comment (justinvdm) [inline molo/profiles/templates/profiles/viewprofile.html:14]: Noting from our discussion so I don't forget: We should not render the mobile number if it is `None`.

File: molo/profiles/tests/test_forms.py [hunk 6]
```diff
@@ -60,3 +73,11 @@ def test_username_exists(self):
...
         self.assertEqual(
             'Username already exists.', validation_error.message)
+
+    def test_terms_and_conditions_is_required(self):
+        form_data = {
+            'username': 'test',
+            'password': '12345',
+        }
+        form = RegistrationForm(data=form_data)
+        self.assertEqual(form.is_valid(), False)
```
Comment (justinvdm) [inline molo/profiles/tests/test_forms.py:83]: Should we add a test for cases where the mobile number field is required.

**Comment by justinvdm** [comment]
Left some comments, looks good.

File: molo/profiles/forms.py (hunk 2 above)
Comment (smarzban) [inline molo/profiles/forms.py:98]: Because it's the edit form, They can either update one of the fields or all them

**Comment by justinvdm** [comment]
:+1: when travis is happy

**Pull request merged**